### PR TITLE
fix(ecs): canContainersAccessInstanceRole is ignored when passed in AsgCapacityProvider constructor

### LINKED ISSUE #20293: aws-ecs: canContainersAccessInstanceRole option is ignored
### Describe the bug

If you set `canContainersAccessInstanceRole: true` on an `AsgCapacityProvider` that setting is ignored if you don't also specify it when calling `addAsgCapacityProvider`.

```typescript
const ingestCapacityProvider = (this.capacityProvider =
  new ecs.AsgCapacityProvider(this, "IngestCapacityProvider", {
    autoScalingGroup,
    enableManagedScaling: true,
    enableManagedTerminationProtection: true,
    // needed so containers can find instance private ip
    canContainersAccessInstanceRole: true, // this is ignored?
  }));

ingestCluster.addAsgCapacityProvider(ingestCapacityProvider, {
  // needed so containers can find instance private ip
  canContainersAccessInstanceRole: true, // this is applied?
});
```

### Expected Behavior

When I set `canContainersAccessInstanceRole: true` on an `AsgCapacityProvider` I expect the resulting `LaunchTemplate`/instances to be allowed to access EC2 metadata service. 

### Current Behavior

When I set `canContainersAccessInstanceRole: true` on an `AsgCapacityProvider` the setting is ignored and the resulting LaunchTemplate/instances block ECS/docker from accessing the EC2 metadata service.

### Reproduction Steps

```typescript
const ingestCapacityProvider = (this.capacityProvider =
  new ecs.AsgCapacityProvider(this, "IngestCapacityProvider", {
    autoScalingGroup,
    enableManagedScaling: true,
    enableManagedTerminationProtection: true,
    // needed so containers can find instance private ip
    canContainersAccessInstanceRole: true, // this is ignored?
  }));

ingestCluster.addAsgCapacityProvider(ingestCapacityProvider, {
  // needed so containers can find instance private ip
  canContainersAccessInstanceRole: true, // this is applied?
});
```

### Possible Solution

Remove the option from `AsgCapacityProvider` since it is ignored anyways. 

### Additional Information/Context

_No response_

### CDK CLI Version

2.23.0 (build 50444aa)

### Framework Version

_No response_

### Node.js Version

v16.13.0

### OS

macos/linux

### Language

Typescript

### Language Version

_No response_

### Other information

_No response_

### PR DESCRIPTION
**Pull request by daschaa**
Fixes #20293 

When adding an AsgCapacityProvider the property `canContainersAccessInstanceRole` is only checked when passed in via the method `addAsgCapacityProvider`. It is ignored when passing the property via the instantiation of an AsgCapacityProvider. In this PR I added, that if either one way (method or constructor) has got the property set - it is respected in the outcome. For more details please see the issue #20293 

I decided **not** to omit the property on the class level because it would bring in breaking changes.

----

### All Submissions:

* [x] Have you followed the guidelines in our [Contributing guide?](https://github.com/aws/aws-cdk/blob/master/CONTRIBUTING.md)

### Adding new Unconventional Dependencies:

* [ ] This PR adds new unconventional dependencies following the process described [here](https://github.com/aws/aws-cdk/blob/master/CONTRIBUTING.md/#adding-new-unconventional-dependencies)

### New Features

* [x] Have you added the new feature to an [integration test](https://github.com/aws/aws-cdk/blob/master/INTEGRATION_TESTS.md)?
	* [x] Did you use `yarn integ` to deploy the infrastructure and generate the snapshot (i.e. `yarn integ` without `--dry-run`)?

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

File: packages/@aws-cdk/aws-ecs/lib/cluster.ts [hunk 1]
```diff
@@ -366,7 +366,10 @@ export class Cluster extends Resource implements ICluster {
     }
     this._hasEc2Capacity = true;
     this.configureAutoScalingGroup(provider.autoScalingGroup, {
-      ...options,
+      ...{
```
Comment (corymhall) [inline packages/@aws-cdk/aws-ecs/lib/cluster.ts:369]:
```suggestion
      {
```

File: packages/@aws-cdk/aws-ecs/lib/cluster.ts [hunk 2]
```diff
@@ -366,7 +366,10 @@ export class Cluster extends Resource implements ICluster {
...
+        ...options,
+        canContainersAccessInstanceRole: provider.canContainersAccessInstanceRole || options.canContainersAccessInstanceRole,
```
Comment (corymhall) [inline packages/@aws-cdk/aws-ecs/lib/cluster.ts:371]:
```suggestion
        canContainersAccessInstanceRole: provider.canContainersAccessInstanceRole ?? options.canContainersAccessInstanceRole,
```

**Review by corymhall — CHANGES_REQUESTED** [review]
@daschaa thanks for the fix!

I'm wondering if it might also be good to throw an error if the user sets `canContainersAccessInstanceRole` on both and they are different values? Like you explicitly set it to `true` on one and `false` on the other, are you sure?

**Comment by daschaa** [comment]
@corymhall Thanks for the hint! I didn't thought about that case. 
An error is maybe a good idea. Another approach/idea is, that `addAsgCapacityProvider` could take precedence over the constructor one. If added explicitly to the method, the property is then "overridden".
What do you think?

**Comment by corymhall** [comment]
> @corymhall Thanks for the hint! I didn't thought about that case. An error is a good idea. But this could maybe lead to breaking changes.
> 
> Another approach/idea is, that `addAsgCapacityProvider` could take precedence over the constructor one. If added explicitly to the method, the property is then "overridden". What do you think?

yeah that's a good call. `addAsgCapacityProvider` should take precedence to keep backwards compatibility 👍

**Comment by daschaa** [comment]
@corymhall Cool! I will add a test for that.

**Comment by daschaa** [comment]
@corymhall I have now added test cases for the precedence. Could you review it again please? 
Thanks for your feedback and help! ❤️

**Review by corymhall — APPROVED** [review]

**Comment by joeflateau** [comment]
Thank you @daschaa!

**Comment by daschaa** [comment]
@joeflateau Thank you for creating the issue! Your finding was 99% of the work ❤️

**Pull request merged**